mParticle-Apple-SDK/Network/MPURLRequestBuilder.m

```diff
@@ -187,33 +203,27 @@ - (NSMutableURLRequest *)build {
             }
             
             signatureMessage = [NSString stringWithFormat:@"%@\n%@\n%@%@", _httpMethod, date, relativePath, _message];
-        } else { // /config and /audience
+        } else { // /config
             contentType = @"application/x-www-form-urlencoded";
             
-            range = [relativePath rangeOfString:@"/config"];
-            if (range.location != NSNotFound) {
-                if (supportedKits) {
-                    kits = [supportedKits componentsJoinedByString:@","];
-                }
-                
-                NSString *environment = [NSString stringWithFormat:@"%d", (int)[MPStateMachine environment]];
-                [urlRequest setValue:environment forHTTPHeaderField:@"x-mp-env"];
-                
-                MPIUserDefaults *userDefaults = [MPIUserDefaults standardUserDefaults];
-                NSString *eTag = userDefaults[kMPHTTPETagHeaderKey];
-                NSDictionary *config = [userDefaults getConfiguration];
-                if (eTag && config) {
-                    [urlRequest setValue:eTag forHTTPHeaderField:@"If-None-Match"];
-                }
-                
-                NSString *query = [_url.defaultURL query];
-                signatureMessage = [NSString stringWithFormat:@"%@\n%@\n%@?%@", _httpMethod, date, relativePath, query];
-            } else {
-                signatureMessage = [NSString stringWithFormat:@"%@\n%@\n%@", _httpMethod, date, relativePath];
+            if (supportedKits) {
+                kits = [supportedKits componentsJoinedByString:@","];
+            }
+            
+            NSString *environment = [NSString stringWithFormat:@"%d", (int)[MPStateMachine environment]];
+            [urlRequest setValue:environment forHTTPHeaderField:@"x-mp-env"];
+            
+            MPIUserDefaults *userDefaults = [MPIUserDefaults standardUserDefaults];
+            NSString *eTag = userDefaults[kMPHTTPETagHeaderKey];
+            NSDictionary *config = [userDefaults getConfiguration];
+            if (eTag && config) {
+                [urlRequest setValue:eTag forHTTPHeaderField:@"If-None-Match"];
             }
+            
+            NSString *query = [_url.defaultURL query];
+            signatureMessage = [NSString stringWithFormat:@"%@\n%@\n%@?%@", _httpMethod, date, relativePath, query];
```

Comment:
should this be in this PR?  i'm unclear as to what is going on here.  zoom might be easier.

Reply:
So when I was testing this existing code with the endpoint it wasn't working because of a url with capitalization issues and because the the signature was malformed. There was also several headers being sent in the audience call that weren't necessary.  So since this was the only ticket I had for this functionality, I fixed it all in this ticket though some of this is also from the changes in that feature branch.